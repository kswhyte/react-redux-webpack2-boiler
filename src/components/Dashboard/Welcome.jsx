import React from 'react';
import PropTypes from 'prop-types';
import HgRow from '@hg/three-ui/HgRow';
import HgButton from '@hg/three-ui/HgButton';

import './welcome.css';

const Welcome = props => {
  return (
    <div className="start-call-banner-block">
      <HgRow className="no-margin">
        <h1>Welcome</h1>
        <hr />
        <HgButton
          className="new-session-button"
          tabIndex={1}
          onClick={props.startSessionClick}
          text="Begin New Session"
        />
        <div className="error-container">
          {(props.startError === 'ERROR') && (<h3>Something went wrong.</h3>)}
          {(props.startError === 'TIMEOUT') && (<h3>Request timed out.</h3>)}
        </div>
      </HgRow>
    </div>
  );
};

Welcome.propTypes = {
    startSessionClick: PropTypes.func,
    startError: PropTypes.bool
};

export default Welcome;
