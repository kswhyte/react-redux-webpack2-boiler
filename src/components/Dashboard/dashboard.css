@import "imports/_helpers.css";

.start-call-banner-block {
  & .error-container {
    width: 100%;
    height: 100%;
    text-align: center;
    padding-top: 200px;
    color: #e37342;
  }
}

.new-session-button {
  font-size: 1.5rem;
  background: var(--primary-color);
  color: white;
}
