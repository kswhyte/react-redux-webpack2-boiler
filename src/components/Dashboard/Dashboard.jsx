import React from 'react';
import PropTypes from 'prop-types';
import HgRow from '@hg/three-ui/HgRow';
import HgButton from '@hg/three-ui/HgButton';
import { withRouter } from 'react-router';
import sStorage from '../../../tools/sessionStorage_helper';

import './dashboard.css';

const Dashboard = props => {
  if(!sStorage.getItem({key:'isUserLoggedIn'}).status){
    props.history.push('/login')
  }
  return (
    <div className="start-call-banner-wrapper">
      <HgRow className="no-margin">
        <h1>Dashboard</h1>
        <hr />
        <HgButton
          className="dashboard-button"
          tabIndex={1}
          onClick={props.startSessionClick}
          text="Begin New Session"
        />
        <HgButton
          className="dashboard-button"
          tabIndex={1}
          text="Manage My Account"
        />
        <div className="error-container">
          {props.startError === 'ERROR' && <h3>Something went wrong.</h3>}
          {props.startError === 'TIMEOUT' && <h3>Request timed out.</h3>}
        </div>
      </HgRow>
    </div>
  );
};

Dashboard.propTypes = {
  startSessionClick: PropTypes.func.isRequired,
  startError: PropTypes.bool,
  history : PropTypes.object.isRequired,
  userLoggedIn: PropTypes.bool.isRequired
};

const DashboardWithRouter = withRouter(Dashboard);

export default DashboardWithRouter;
