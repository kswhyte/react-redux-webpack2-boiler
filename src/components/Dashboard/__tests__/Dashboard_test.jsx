import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import {Dashboard} from '../Dashboard';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { shallow, mount } from 'enzyme';
import expect from 'expect';

const createCall = () => {
  console.log('Beginning the call');
};

const startProps = {
  createCall,
  startError: null,
  startSessionClick: function(){},
  history: {},
  userLoggedIn: true
};

const errorProps = {
  createCall,
  startError: 'ERROR',
  startSessionClick: function(){},
  history: {},
  userLoggedIn: true
};

const timeoutProps = {
  createCall,
  startError: 'TIMEOUT',
  startSessionClick: function(){},
  history: {},
  userLoggedIn: true
};
/* may remove this */
const jest = jest || null;
if(jest){
  window.sessionStorage = {
    setItem: function(){},
    getItem: function(){},
    removeItem: function(){}
  };
}

const stories = storiesOf('Dashboard', module);

stories.add('Dashboard page main story', () => {
  const dashboardStory = <Dashboard {...startProps} />;
  specs(() =>
    describe('Show the main start call page', () => {
      it('Show the main start call page', () => {
        let output = shallow(dashboardStory);
        expect(output.find('h2').text()).toEqual('Dashboard');
      });
    })
  );
  return dashboardStory;
});

stories.add('Dashboard page main error story', () => {
  const dashboardStory = <Dashboard {...errorProps} />;
  specs(() =>
    describe('Show a generic somethine went wrong error', () => {
      it('Show a generic somethine went wrong error', () => {
        let output = shallow(dashboardStory);
        expect(output.find('h3').first().text()).toContain('Something went wrong');
      });
    })
  );
  return dashboardStory;
});

stories.add('Dashboard page timeout error story', () => {
  const dashboardStory = <Dashboard {...timeoutProps} />;
  specs(() =>
    describe('Show a timeout error', () => {
      it('Show a timeout error', () => {
        let output = shallow(dashboardStory);
        expect(output.find('h3').first().text()).toContain('Request timed out');
      });
    })
  );
  return dashboardStory;
});
