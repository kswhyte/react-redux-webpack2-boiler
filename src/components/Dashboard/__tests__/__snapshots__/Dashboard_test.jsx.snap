// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page main error story 1`] = `
<div
  className="start-call-banner-wrapper"
>
  <div
    className="no-margin"
  >
    <h2
      className="dashboard-text"
    >
      Dashboard
    </h2>
    <hr />
    <button
      className="new-session-button primary"
      onClick={[Function]}
      tabIndex="1"
    >
      Begin New Session
    </button>
    <div
      className="error-container"
    >
      <h3>
        Something went wrong.
      </h3>
    </div>
  </div>
</div>
`;

exports[`Dashboard page main story 1`] = `
<div
  className="start-call-banner-wrapper"
>
  <div
    className="no-margin"
  >
    <h2
      className="dashboard-text"
    >
      Dashboard
    </h2>
    <hr />
    <button
      className="new-session-button primary"
      onClick={[Function]}
      tabIndex="1"
    >
      Begin New Session
    </button>
    <div
      className="error-container"
    />
  </div>
</div>
`;

exports[`Dashboard page timeout error story 1`] = `
<div
  className="start-call-banner-wrapper"
>
  <div
    className="no-margin"
  >
    <h2
      className="dashboard-text"
    >
      Dashboard
    </h2>
    <hr />
    <button
      className="new-session-button primary"
      onClick={[Function]}
      tabIndex="1"
    >
      Begin New Session
    </button>
    <div
      className="error-container"
    >
      <h3>
        Request timed out.
      </h3>
    </div>
  </div>
</div>
`;
